.github/workflows/test.yml

```diff
@@ -0,0 +1,27 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v4
```

Comment:
sorry I didn't notice this on the first pass, but you should be using `actions/setup-node@v4` with `cache: pnpm` set. You need to do this after `pnpm/action-setup@v4` for it to work right 👍